### PR TITLE
Save/ Unsave #36 issue fixed

### PR DESCRIPTION
**Pull request by vasanthakumar-a**
(Save/Unsave)[https://github.com/MilesUpHQ/reddit-clone-rails/issues/36] fixed
Using ajax patch request is sent and saved is added to posts

### CODE REVIEW

File: app/controllers/save_controller.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      @post.update(saved: false)
+    else
+      @post.update(saved: true)
+    end
+
+    respond_to do |format|
+      format.js {
+        @post = Post.find(params[:id])
+        @is_saved = @post.saved
+        render "save/create"
```
Comment (iMagesh) [inline app/controllers/save_controller.rb:15]: There seems to be a few duplicates in here, can we refactor?

File: app/helpers/save_helper.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module SaveHelper
+
+  def is_saved post
+    account_signed_in? && post.saved === "true" ? " active" : ""
```
Comment (iMagesh) [inline app/helpers/save_helper.rb:4]: can we refactor this?

File: app/javascript/includes/vote.js [hunk 3]
```diff
@@ -1,10 +1,10 @@
 $( document ).on('turbolinks:load', function() {
   $(".vote").on("click", ".upvote, .downvote", function(){
-    var post_id = $(this).parent().data("id"),
+    let post_id = $(this).parent().data("id"),
       is_upvote = $(this).hasClass("upvote");
```
Comment (iMagesh) [inline app/javascript/includes/vote.js:4]: use let for is_upvote as well?

File: app/javascript/packs/application.js [hunk 4]
```diff
@@ -11,6 +11,7 @@ require("channels")
 require("bootstrap")
 
 require("../includes/vote.js")
+require("../includes/save.js")
```
Comment (iMagesh) [inline app/javascript/packs/application.js:14]: this file name "save" seems to be very generic, we need to know what we are saving..
Can we change the name to make it clear?

File: app/controllers/save_controller.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
+class SaveController < ApplicationController
```
Comment (iMagesh) [inline app/controllers/save_controller.rb:1]: SaveController seems to be a very generic name. Can we be a little specific to not confuse others? can we choose a better name so it is clear to others on what we are doing within this file?

File: app/helpers/save_helper.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+module SaveHelper
```
Comment (iMagesh) [inline app/helpers/save_helper.rb:1]: same for this as well. The name should be specific.

**Review by iMagesh — CHANGES_REQUESTED** [review]
I have added comments please check and update.

File: app/controllers/save_controller.rb (hunk 1 above)
Comment (vasanthakumar-a) [reply app/controllers/save_controller.rb]: Resolved it

File: app/helpers/save_helper.rb (hunk 2 above)
Comment (vasanthakumar-a) [reply app/helpers/save_helper.rb]: Resolved it

File: app/javascript/packs/application.js (hunk 4 above)
Comment (vasanthakumar-a) [reply app/javascript/packs/application.js]: Resolved it

File: app/controllers/save_controller.rb (hunk 5 above)
Comment (vasanthakumar-a) [reply app/controllers/save_controller.rb]: Resolved it

File: app/helpers/save_helper.rb (hunk 6 above)
Comment (vasanthakumar-a) [reply app/helpers/save_helper.rb]: Resolved it

File: app/javascript/includes/vote.js (hunk 3 above)
Comment (vasanthakumar-a) [reply app/javascript/includes/vote.js:4]: already it is defined with 'let'

**Comment by vasanthakumar-a** [comment]
All comments resolved and its now ready to merge

**Pull request merged**